testsuite/features/support/env.rb

```diff
@@ -127,31 +127,59 @@ def capybara_register_driver
   $feature_scope = scenario.location.file.split(%r{(\.feature|/)})[-2]
 end
 
-# embed a screenshot after each failed scenario
+# Embed a screenshot after each failed scenario
 After do |scenario|
   current_epoch = Time.new.to_i
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
-  if scenario.failed?
+
+  return unless scenario.failed?
+
+  Dir.mkdir('screenshots') unless File.directory?('screenshots')
+  path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
+
+  if web_session_is_active?
+    handle_screenshot_and_relog(path)
+  else
+    warn 'Page is not visible; unable to take a screenshot.'
+  end
+
+  page.instance_variable_set(:@touched, false)
+end
+
+def web_session_is_active?
+  # Test is web session is open
+  page.has_selector?('header') || page.has_selector?('#username-field')
```

Comment:
I removed the wait time for this test. It could be risky but will speedup a lot the failing scenario without web session open. For me, if we reach this point, it means our scenario already failed on a timeout if we have a web session active.

Former code : 
```ruby
def web_session_is_active?
  page.has_selector?('header', wait: Capybara.default_max_wait_time) ||
    page.has_selector?('#username-field', wait: Capybara.default_max_wait_time)
end
```